Feedback

Gildoniel opened the pull request:
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](https://classroom.github.com/help/leaving-feedback-in-github)”.
</details>


Subscribed: @Schweinschwaller

Gildoniel committed:
Setting up GitHub Classroom Feedback

Schweinschwaller committed:
Add exercise 8.3

Schweinschwaller committed:
Add Black Jack

Schweinschwaller committed:
Add Black Jack

Schweinschwaller committed:
black jack still a bit bugy

Schweinschwaller committed:
Black jack cardvalue count bug fix with sort and reorganisation

Schweinschwaller committed:
Using now my own version of Scanner

Schweinschwaller committed:
Add my own version of Scanner

Schweinschwaller committed:
Add nextYN and nextBoolean to GScanner

Schweinschwaller committed:
Add some Comments

gerhofer commented on src/GScanner.java:
```diff
@@ -0,0 +1,83 @@
+        s = new Scanner(System.in);
+    }
+
+    public int nextInt() {
```
mcaht diese methode irgendwas anders als die klassische nextInt() methode? 🤔

Schweinschwaller replied:
ja wenn ich mich vertippe dann ist dass program nicht beendet ;) 
wenn ich 10w eingebe bei standart nextInt dann bekomm ich:
Exception in thread "main" java.util.InputMismatchException
	at java.base/java.util.Scanner.throwFor(Scanner.java:939)
	at java.base/java.util.Scanner.next(Scanner.java:1594)
	at java.base/java.util.Scanner.nextInt(Scanner.java:2258)
	at java.base/java.util.Scanner.nextInt(Scanner.java:2212)
	at com.codersbay.Main.main(Main.java:X)

bei meinen nextInt bekomm ich 10 und das program ist nicht beendet.
ich hab mir leichter beim testen getan. besonders weil ich das menü auch mit zahlen verwende.

gerhofer commented on src/Roulette.java:
```diff
@@ -0,0 +1,338 @@
+        GScanner gs = new GScanner();
+        int money = gs.nextInt();
+        if (money > 100) {
+            System.out.println("☺");
```
🤣

gerhofer commented on src/Roulette.java:
```diff
@@ -0,0 +1,338 @@
+
+        System.out.println("\n".repeat(20));
+        System.out.println("##########################################################");
+        System.out.println("#                Wellcome to ROULETTE                    #");
```
Welcome*  😉

gerhofer commented on src/Roulette.java:
```diff
@@ -0,0 +1,338 @@
+            int chosenNumber = 0;
+            if (menu == 6) {
+                do {
+                    System.out.println("Witch number?");
```
`Which` wir wollen keine Hexen 🤓

gerhofer commented on src/Roulette.java:
```diff
@@ -0,0 +1,338 @@
+                do {
+                    System.out.println("Witch number?");
+                    chosenNumber = gs.nextInt();
+                    if (setCommitment == -1) {
```
wieso machen wir die setCommitment abfrage nur bei menu = 6 ?

Schweinschwaller replied:
ups da gehört menü hin.
wenn man -1 eingibt dann geht man wieder raus

gerhofer commented on src/Roulette.java:
```diff
@@ -0,0 +1,338 @@
+                money += setCommitment * 3;
+            } else if (menu == 1 && fortunaChoice % 2 != 0) {
+                money += setCommitment * 2;
+            } else if (menu == 2 && fortunaChoice % 2 == 0) {
```
krieg ich jetzt wenn ich auf schwarz setze auch geld wenn eine null kommt 😱 🤔 das wär schlecht für dein casion 😉

gerhofer commented on src/Roulette.java:
```diff
@@ -0,0 +1,338 @@
+            } else if (cards.get(i) > 7 && cards.get(i) <= 11) {
+                value += 10;
+            } else {
+                if (value <= 10) {
```
hier vllt einen kommentar was diese logik ausdrückt, ich verstehs jetzt so aus dem nichts eigentlich nicht 😆

Schweinschwaller replied:
public static int cardValue(ArrayList<Integer> cards) {
        // in Black Jack the Cards from 2 until 10 have the value of it self.
        // J D K have the value of 10
        // the ace have the value of 1 or 11
        // there are 13 cards 2,3,4,5,6,7,8,9,10,J,D,K,♦
        int value = 0;
        // with sort of the cards a ace will be at the and of the cards
        Collections.sort(cards);
        for (int i = 0; i < cards.size(); i++) {
            //draw a Card gives a number [0-12] => 0 == |2| 1 == |3| 2 == |4| ....
            if (cards.get(i) >= 0 && cards.get(i) <= 7) {
                // the random number +2 is the value of the card
                value += cards.get(i) + 2;
                // 10,J,D,K add 10 to the value
            } else if (cards.get(i) > 7 && cards.get(i) <= 11) {
                value += 10;
                // the ace!!! can have the value of 1 or 11
            } else {
                if (value <= 10) {
                    value += 11;
                } else {
                    value += 1;
                }
            }
        }
        return value;
    }

Schweinschwaller committed:
Bugfix natural Black Jack

gerhofer commented on src/GScanner.java:
```diff
@@ -0,0 +1,83 @@
+        return ' ';
+    }
+
+    public boolean nextYN() {
```
cool 🤓

gerhofer reviewed (changes requested):
richtig, richtig cool 👍 

was noch super wär, wär wenn du a bisserl probierst, dass du kleinere Methoden extrahierst um deinen ode möglichst lesbar zu schreiben. Algorithmisch programmierst du schon super gut, jetzt könntest du dich a bisserl drauf konzentrieren den Code so leserlich wie möglich zu gestalten 😉

Schweinschwaller committed:
Changes because of the Feedback
Roulett special fall 0 by wincondition changed
Black Jack cardValue comment
and some small changes of the output on the consol

Schweinschwaller committed:
comments added

DaStanzel commented:
http://gph.is/1W4D3Zo
